src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -193,11 +193,19 @@
      * Use MixpanelAPI.getInstance to get an instance.
      */
     MixpanelAPI(Context context, Future<SharedPreferences> referrerPreferences, String token) {
+        this(context, referrerPreferences, token, MPConfig.getInstance(context));
+    }
+
+    /**
+     * You shouldn't instantiate MixpanelAPI objects directly.
+     * Use MixpanelAPI.getInstance to get an instance.
+     */
+    MixpanelAPI(Context context, Future<SharedPreferences> referrerPreferences, String token, final MPConfig config) {
```

Comment:
For consistency, we can just remove `final` modifier. Alternatively, we can add `final` to the rest of the constructor params.